pkg/controllers/cloud/cloudentityselector_controller.go

```diff
@@ -153,12 +153,21 @@ func (r *CloudEntitySelectorReconciler) processCreateOrUpdate(selector *cloudv1a
 	r.selectorToAccountMap[*selectorNamespacedName] = *accountNamespacedName
 	err = cloudInterface.AddAccountResourceSelector(accountNamespacedName, selector)
 	if err != nil {
+		r.Log.Info(errorMsgSelectorAddFail, "selector", selectorNamespacedName, "error", err)
 		_ = r.processDelete(selectorNamespacedName)
 		return fmt.Errorf("%s %s, err: %v", errorMsgSelectorAddFail, selectorNamespacedName.Name, err)
 	}
 
 	err = r.Poller.updateAccountPoller(accountNamespacedName, selector)
 	if err != nil {
+		r.Log.Info(errorMsgSelectorAddFail, "selector", selectorNamespacedName, "error", err)
+		_ = r.processDelete(selectorNamespacedName)
+		return err
+	}
+
+	err = r.Poller.RestartAccountPoller(accountNamespacedName)
+	if err != nil {
+		r.Log.Info(errorMsgSelectorAddFail, "selector", selectorNamespacedName, "error", err)
 		_ = r.processDelete(selectorNamespacedName)
```

Comment:
As this is called twice, can we use defer function for error handling

Reply:
Will let @archanapholla  address it.

Reply:
addressed.